src/Xamarin.SourceWriter/Models/MethodWriter.cs

```diff
@@ -52,11 +52,16 @@ public void SetVisibility (string visibility)
 
 		public virtual void Write (CodeWriter writer)
 		{
+			WritePartialMethodDeclaration (writer);
```

Comment:
This way of implementation breaks the "spirit" of the SourceWriter model.  That is, it is using a `MethodWriter` to write 2 methods.

This should instead be implemented as a new `ConstructorPartialMethod : MethodWriter` class in `generator.csproj` that gets added to the `ClassWriter` model when the `JavaLangObjectConstructor` is created.

ie, here:
https://github.com/jonpryor/java.interop/blob/jonp-ctor-calls-partial-method/tools/generator/SourceWriters/BoundClass.cs#L128-L130

Should be something like:
```csharp
// Add required constructor for all JLO inheriting classes
if (klass.FullName != "Java.Lang.Object" && klass.InheritsObject) {
	Methods.Add (new ConstructorPartialMethod (klass, opt));
	Constructors.Add (new JavaLangObjectConstructor (klass, opt));
}
```

The only change we should need for `MethodWriter` is if it doesn't support partial methods in general.